stackdriver/config/config.go

```diff
@@ -0,0 +1,128 @@
+	// service account key to use.
+	secretDataFieldKey = "key.json"
+
+	// ProjectIDKey is the name of the ConfigMap field for ProjectID.
+	ProjectIDKey = "project-id"
+	// GcpLocationKey is name of the ConfigMap field for GcpLocation.
+	GcpLocationKey = "gcp-location"
+	// ClusterNameKey is the name of the ConfigMap field for ClusterName.
+	ClusterNameKey = "cluster-name"
+	// GcpSecretNameKey is the name of the ConfigMap field for GcpSecretName.
+	GcpSecretNameKey = "gcp-secret-name"
+)
+
+// Config encapsulates the metadata required to configure a Stackdriver client.
+type Config struct {
+	// ProjectID is the stackdriver project ID to which data is uploaded.
+	// This is not necessarily the GCP project ID where the Kubernetes cluster is hosted.
+	// Required when the Kubernetes cluster is not hosted on GCE.
+	ProjectID string
+	// GcpLocation is the GCP region or zone to which data is uploaded.
+	// This is not necessarily the GCP project ID where the Kubernetes cluster is hosted.
+	// Required when the Kubernetes cluster is not hosted on GCE.
+	GcpLocation string
+	// ClusterName is the cluster name with which the data will be associated in Stackdriver.
+	// Required when the Kubernetes cluster is not hosted on GCE.
+	ClusterName string
+	// GcpSecretName is the optional GCP service account key which will be used to
+	// authenticate with Stackdriver. If not provided, Google Application Default Credentials
+	// will be used (https://cloud.google.com/docs/authentication/production).
+	GcpSecretName string
+}
+
+// NewStackdriverConfigFromConfigMap returns a Config for the given configmap
+func NewStackdriverConfigFromConfigMap(config *corev1.ConfigMap) (*Config, error) {
+	return NewStackdriverConfigFromMap(config.Data)
+}
+
+// NewStackdriverConfigFromMap returns a Config for the given map
+func NewStackdriverConfigFromMap(config map[string]string) (*Config, error) {
+	sc := &Config{}
+
+	if pi, ok := config[ProjectIDKey]; ok {
+		sc.ProjectID = pi
+	}
+
+	if gl, ok := config[GcpLocationKey]; ok {
+		sc.GcpLocation = gl
+	}
+
+	if cn, ok := config[ClusterNameKey]; ok {
+		sc.ClusterName = cn
+	}
+
+	if gsn, ok := config[GcpSecretNameKey]; ok {
+		sc.GcpSecretName = gsn
+	}
+
+	return sc, nil
+}
+
+var (
+	// kubeclient is the in-cluster Kubernetes kubeclient.
+	kubeclient *kubernetes.Clientset
+)
+
+// GetStackdriverSecret returns the Kubernetes Secret specified in the given config.
+func GetStackdriverSecret(config *Config) (*v1.Secret, error) {
+	ensureKubeclient()
+	return kubeclient.CoreV1().Secrets(secretNamespace).Get(config.GcpSecretName, metav1.GetOptions{})
+}
+
+// ConvertSecretToExporterOption converts a Kubernetes Secret to an OpenCensus Stackdriver Exporter Option.
+func ConvertSecretToExporterOption(secret *v1.Secret) option.ClientOption {
+	return option.WithCredentialsJSON(secret.Data[secretDataFieldKey])
+}
+
+func ensureKubeclient() {
+	if kubeclient != nil {
```

Comment:
Do we need to add lock to make sure reading this var is multiple-threads safe?

Reply:
Yes, that's probably a good idea.

Reply:
stackdriver/config changes moved to #789